### PR TITLE
feat: add cloudflared system extension

### PR DESCRIPTION
**Pull request by maxnrb**
This PR adds a new system extension cloudflared

Cloudflare Tunnel securely connects resources to Cloudflare without a public IP. A lightweight daemon (cloudflared) creates outbound-only connections to Cloudflare, allowing safe access to services like HTTP, SSH, remote desktops, and other protocols.

More info: https://github.com/cloudflare/cloudflared/

### CODE REVIEW

File: network/cloudflared/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  version: "$VERSION"
+  author: Maxime Narbaud
+  description: |
+    Cloudflare Tunnel securely connects resources to Cloudflare without a public IP.
+    A lightweight daemon (cloudflared) creates outbound-only connections to Cloudflare,
+    allowing safe access to services like HTTP, SSH, remote desktops, and other protocols.
+    More info: https://github.com/cloudflare/cloudflared/
+  compatibility:
+    talos:
+      version: ">= v1.5.0"
```
Comment (frezbo) [inline network/cloudflared/manifest.yaml:13]: 1.9.0

Comment (smira) [reply network/cloudflared/manifest.yaml:13]: I think from pure API perspective, 1.5.0 should be fine as well? (if someone builds manually?)

Comment (frezbo) [reply network/cloudflared/manifest.yaml:13]: yeh, will keep as is then

**Review by frezbo — APPROVED** [review]

**Comment by maxnrb** [comment]
Hi @frezbo, 

I've just seen your commit, and instead of retrieving the cloudflared binaries directly, you locate the sources and build the binary. 

However, cloudflared is built with a version of go ‘modified’ by Cloudflare (https://github.com/cloudflare/go), see the Dockerfile: https://github.com/cloudflare/cloudflared/blob/master/Dockerfile
So I'm afraid that some of the tunnel's functionality won't work (particularly security functionality, as their version patches the TLS stack).

I can push a new PR by building the binary from source with Cloudflare's go executable. What do you think about?

**Comment by frezbo** [comment]
> Hi @frezbo,
> 
> I've just seen your commit, and instead of retrieving the cloudflared binaries directly, you locate the sources and build the binary.

We usually do that, so we build all binaries with the go version that ships with our toolchain, we pick a different one only for special cases

> 
> However, cloudflared is built with a version of go ‘modified’ by Cloudflare (https://github.com/cloudflare/go), see the Dockerfile: https://github.com/cloudflare/cloudflared/blob/master/Dockerfile So I'm afraid that some of the tunnel's functionality won't work (particularly security functionality, as their version patches the TLS stack).

Are you sure https://github.com/cloudflare/cloudflared/blob/master/Dockerfile#L4 this just seems to use upstream golang image

**Comment by frezbo** [comment]
> Are you sure https://github.com/cloudflare/cloudflared/blob/master/Dockerfile#L4 this just seems to use upstream golang image

nevermind, saw the lines after that, in that case I guess we can fallback to pre-built binaries, wdyt @smira

**Comment by smira** [comment]
> > Are you sure [cloudflare/cloudflared@`master`/Dockerfile#L4](https://github.com/cloudflare/cloudflared/blob/master/Dockerfile?rgh-link-date=2024-12-12T03%3A57%3A12Z#L4) this just seems to use upstream golang image
> 
> nevermind, saw the lines after that, in that case I guess we can fallback to pre-built binaries, wdyt @smira

idk, mixed feelings about that. but I guess if someone runs `cloudflared`, they already trust their security to Cloudflare, so using upstream binaries is fine.

**Comment by maxnrb** [comment]
Hi, so finally ?

**Comment by frezbo** [comment]
> Hi, so finally ?

yes, we would use the binaries

**Comment by maxnrb** [comment]
Okay, I'll push a fix then

**Pull request merged**